### PR TITLE
Add the ability to start a netcore based plugin

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Fixes: https://github.com/NuGet/Home/issues/6706
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: 

Add the ability to start plugins cross-plat. 

There is really no good way to automate testing this, since a plugin need to be authenticode signed to be started.

## Testing/Validation
Tests Added: No  
Reason for not adding tests:  
Validation done:  
I've done quite a bit of manual validation with the following plugin https://github.com/nkolev92/NuGet.Tools/tree/master/NuGet.CredentialProvider.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs [hunk 1]
```diff
@@ -144,16 +144,29 @@ public void Dispose()
...
                 RedirectStandardError = false,
                 RedirectStandardInput = true,
                 RedirectStandardOutput = true,
                 StandardOutputEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false)
             };
-
+#else
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH"),
```
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs:161]: What is this usually set to?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs:161]: The dotnet.exe full path.

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs [hunk 2]
```diff
@@ -144,16 +144,29 @@ public void Dispose()
...
+                Arguments = $"\"{filePath}\" " + args,
+                UseShellExecute = false,
+                RedirectStandardError = false,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs:166]: Where are you reading the standard output?

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs [hunk 3]
```diff
@@ -144,16 +144,29 @@ public void Dispose()
...
                 StandardOutputEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false)
             };
-
+#else
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH"),
+                Arguments = $"\"{filePath}\" " + args,
+                UseShellExecute = false,
+                RedirectStandardError = false,
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs:164]: Why is this different from standard output?

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs (hunk 2 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs:166]: The contract for the plugins is that they communicate with NuGet via STDIN/SDTOUT.

File: src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs (hunk 3 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs:164]: https://github.com/NuGet/NuGet.Client/pull/2211/files#r186987163

stderr is ignored by design.

**Comment by nkolev92** [comment]
🔔 @rohit21agrawal @mishra14

**Review by mishra14 — APPROVED** [review]

**Review by rohit21agrawal — APPROVED** [review]

**Review by jainaashish — APPROVED** [review]
LGTM

**Pull request merged**